sklearn/gaussian_process/kernels.py

```diff
@@ -1787,3 +1787,177 @@ def is_stationary(self):
     def __repr__(self):
         return "{0}(gamma={1}, metric={2})".format(
             self.__class__.__name__, self.gamma, self.metric)
+
+
+class SelectDimensionKernel(Kernel):
+    """Wrapper for applying kernels on selected active dimensions of input data.
+
+
+    Parameters
+    ----------
+    kernel : a kernel object.
+        an instantiation of kernel object.
+
+    active_dims : array-like, shape (n_active_dims,)
+
+    """
+
+    def __init__(self, kernel, active_dims):
+        self.active_dims = active_dims
+        self.kernel = kernel
+
+    def __call__(self, X, Y=None, eval_gradient=False):
+        """Return the kernel k(X, Y) and optionally its gradient on selected
+           dimensions.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples_X, n_features)
+            Left argument of the returned kernel k(X, Y)
+
+        Y : array, shape (n_samples_Y, n_features), (optional, default=None)
+            Right argument of the returned kernel k(X, Y). If None, k(X, X)
+            evaluated instead.
+
+        eval_gradient : bool (optional, default=False)
+            Determines whether the gradient with respect to the kernel
+            hyperparameter is determined. Only supported when Y is None.
+
+        Returns
+        -------
+        K : array, shape (n_samples_X, n_samples_Y)
+            Kernel k(X[:,active_dims], Y[:,active_dims])
+
+        K_gradient : array (opt.) shape (n_samples X n_samples X active_dims)
+            The gradient of the kernel k(X, X) with respect to the
+            hyperparameter of the kernel. Only returned when eval_gradient
+            is True.
+        """
+
+        active_dims = np.asarray(self.active_dims, dtype=np.int)
```

Comment:
Implementation is straight forward, my concern was that it may be confusing for users to undrestand. I will implement it and then maybe you can suggest the wording of doc string.